src/node.cpp

```diff
@@ -1443,19 +1442,11 @@ treestate_t LocalNode::checkstate()
 
 void LocalNode::setnode(Node* cnode)
 {
-    if (node && (node != cnode) && node->localnode)
-    {
-        node->localnode = NULL;
-    }
-
     deleted = false;
 
-    node = cnode;
-
-    if (node)
-    {
-        node->localnode = this;
-    }
+    if (cnode) cnode->localnode.reset();
```

Comment:
I'm not 200% sure if if's feasible, but we might want to check for (LocalNode*)~0:
```suggestion
    if (cnode) 
    {
        if (cnode->localnode.get() == (LocalNode*)~0)
        {
            cnode->localnode.store_unchecked(nullptr);
        }
        else
        {
            cnode->localnode.reset();
        }
    }
```

Reply:
We've prevented the ~0 leaving the function that uses that idea, so I don't think we need this change - if it does occur it's surely a bad memory access and we'll know from crash stacks.  And also, with the upcoming sync rework these pointers go away anyway.